### PR TITLE
Release proposal

### PR DESCRIPTION
**Pull request by atakanyenel**
about #86 . This proposal is WIP and is subject to change.

### CODE REVIEW

File: Release.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# How Release Pipeline should work
+
+Information:
+
+- The version of the release is determined by package.json. When `npm publish` is called, It tries to create a new release with that version.
+
+- `npm version patch` increases the patch number of the package.json. This change should be commited.
```
Comment (a-thaler) [inline Release.md]: am not sure .. changes will trigger a pipeline always. can we skip that for now?

File: Release.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- We don't want to create a new release everytime we push to master.
+
+# Proposal
+1) We create a new branch called `release`. Release branch will only be updated when we have a new release. It will be only merged from master , **not from any other branch.**
+
+2) We create a new job in Prow for release branch for PR coming from master. `pre-submit-release-job` runs the tests.
+
+3) Once PR is merged to `release` branch, a `post-submit-release-job` is triggered. 
+
+- It updates the version with `npm vervion`.
```
Comment (a-thaler) [inline Release.md]: As mentioned above, can be a manual action for now as part of PR changes

File: Release.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+- Publishes the package with `npm publish`.
+- Once publish is complete, kyma-bot merges release branch with master so that master branch also has the updated version number.
```
Comment (a-thaler) [inline Release.md]: I think that part will be tricky and is somehow against the typical branching strategy. When working with develop/master branches you usually merge only in one direction.

Can't we keep the versions stable for now in the master branch? So never touch them and use dummy version numbers like "develop". Then there is no need to merge back

**Comment by obadrawi** [comment]
It looks like a promising approach if writing to the package.json works fine

File: Release.md (hunk 3 above)
Comment (obadrawi) [reply Release.md]: So the master branch doesn't that way represent the latest?

**Comment by a-thaler** [comment]
Please assure that prow will execute an npm publish whenever the release branch gets changes, the credentials for the publish should not be located on local machines

**Pull request merged**